src/app/child-dev-project/reporting/aggregation.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from "@angular/core";
+import { EntityMapperService } from "../../core/entity/entity-mapper.service";
+import { Entity } from "../../core/entity/entity";
+import { Child } from "../children/model/child";
+import { School } from "../schools/model/school";
+import { Note } from "../notes/model/note";
+import { RecurringActivity } from "../attendance/model/recurring-activity";
+import { EventNote } from "../attendance/model/event-note";
+
+export interface PropertyQuery {
+  [key: string]: any;
+}
+
+@Injectable({
+  providedIn: "root",
+})
+export class AggregationService {
+  private static readonly ENTITY_CLASSES: typeof Entity[] = [
+    Child,
+    School,
+    RecurringActivity,
+    Note,
+    EventNote,
+  ];
+
+  private entities = new Map<typeof Entity, Entity[]>();
+  private dataLoaded: Promise<any>;
+  constructor(private entityMapper: EntityMapperService) {}
+
+  public loadData(): void {
+    this.dataLoaded = Promise.all(
+      AggregationService.ENTITY_CLASSES.map((entityClass) =>
```

Comment:
Do we have safeguards if the data should ever be larger than available memory? Are you testing this with (pouchdb) demo data of >1000 students, >5 years of activity data?